How to automate your github workflow and go home early (#3)
MyOnlineStore/blogs: how-to-go-home-early-github.md

To set this up, go to your GitHub repo with actions enabled, click on "New workflow" and once in the marketplace scroll down to "Automate every step in your process" where you'll find the "labeler" action:

![Alt screenshot of GitHub action in marketplace](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/labeler-marketplace.png)

### Automatically request reviews with "Code owners" 

We can take the automatic labeling of your PR one step further. There is a feature in GitHub called "Code owners" which let's you assign co-workers to certain parts of your application(s) GitHub will use codeowners to automatically asign reviews.

![Alt screenshot of reviewers being assigned](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/codeowners.png)

Setting up code owners is easy, check [this guide](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) to get started with code owners.

### Let a panda spam your co-workers into submission for reviews.

> One thing that helped our team a lot was the addition of [Pull panda](https://pullreminders.com/). It is a slack integration that automatically sends a message to your coworkers when someone requests their review. 

Reviewer: Use of 'co-workers' and 'coworkers'. Both are fine (a.f.a.i.k.) but I think the consistent use of either one is better.

> One thing that helped our team a lot was the addition of [Pull panda](https://pullreminders.com/). It is a slack integration that automatically sends a message to your coworkers when someone requests their review. 

Reviewer: 'slack integration' should 'Slack integration'

When you request a review, the reviewer gets a message like:
![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-assigned.png)

If someone approves your pull request, you'll be notified as well:
![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-approved.png)

Even if someone comments on your pull request, panda will show you the comments right away in slack:

![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-comments.png)

> ### Automerge the pull request

Reviewer: 'Automerge', is it 'to automatically merge'? Then it should be 'Auto-merge'. If it's Automerge by Oracle or a function name, keep it like this.

We can even automate the merging of our pull request with another GitHub action called [automerge](https://github.com/pascalgn/automerge-action). This action automatically merges your pull request based on your team's preferences.

![Alt screenshot of GitHub merging the pull request](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/automerge.png)

We configured it so that it merges a pull request once:
- An "automerge" label is added to the pull request
- All required checks are complete
- All the required reviews are approved

And of course after your pull request is automatically merged, you get a message from pull panda:

![Alt screenshot of pull panda automerge message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-automerge.png)

### Conclusion

> I hope this list gave you some inspiration on how automate the little things in your workflow. Even though these steps are probably not the worst bottlenecks for the speed of your team, saving a few minutes per pull request can quickly add up to something worthwhile.

Reviewer: The first sentence is a little bit off. I feel like this should be 'I hope this list gives you some inspiration on how **to** automate [...]'

# How to automate your workflow and go home early (Pull-requests)

_This is a blog series dedicated to becoming the fastest version of yourself by automating tasks that keep you away from actually writing code. I will talk you through how we keep our developers fast and happy at MyOnlineStore._

> ## Github

Reviewer: This is somewhat debatable. With MyOnlineStore you chose to write it stylized, so why not do the same with GitHub wherever the branding is mentioned?

> If you have more than 0 co-workers you are probably using some kind of platform to help you with pull-requests and reviews. If that platform of choice is Github, you're in luck. We compiled a list of tools and tricks that can help you get your commits merged in that sweet master branch as fast as possible.

Reviewer: _"If you have more than 0 co-workers"_ I suggest writing out numbers; 'zero'.
_"some kind of platform to help you with pull-requests"_ pull request (both singular and plural) are actually two words.

> Writing a good pull request template can help you and teammates find a common ground on how to summarize what's in a PR. You can also add add checklist if there are some requirements that have to be met before it can be merged.

Reviewer 2: ```suggestion
Writing a good pull request template can help you and teammates find a common ground on how to summarize what's in a PR. You can also add a checklist if there are some requirements that have to be met before it can be merged.
```

![Alt screenshot of example template](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/pr-template.png)

### Auto label your pull requests

> With the new GitHub actions you can automatically label your pull requests based on what files you touched in your PR. This can help a lot with the discoverability of your pull request overview.

Reviewer 2: GitHub stylizes this as a brand name i.e., `GitHub Actions`.

![Alt screenshot of GitHub action on timeline](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/labeler-timeline.png)

To set this up, go to your GitHub repo with actions enabled, click on "New workflow" and once in the marketplace scroll down to "Automate every step in your process" where you'll find the "labeler" action:

![Alt screenshot of GitHub action in marketplace](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/labeler-marketplace.png)

### Automatically request reviews with "Code owners" 

> We can take the automatic labeling of your PR one step further. There is a feature in GitHub called "Code owners" which let's you assign co-workers to certain parts of your application(s) GitHub will use codeowners to automatically asign reviews.

Reviewer 2: ```suggestion
We can take the automatic labeling of your PR one step further. There is a feature in GitHub called "codeowners" which let's you assign co-workers to certain parts of your application(s). GitHub will use codeowners to automatically assign reviews.
```
or alternatively
```suggestion
We can take the automatic labeling of your PR one step further. There is a feature in GitHub called "Code owners" which let's you assign co-workers to certain parts of your application(s). GitHub will use "Code owners" to automatically assign reviews.
```
or 
```suggestion
We can take the automatic labeling of your PR one step further. There is a feature in GitHub called "code owners" which let's you assign co-workers to certain parts of your application(s). GitHub will use code owners to automatically assign reviews.
```
As long as its consistent 😄. In any case the `assign` typo and period after `application(s)` should be added.

![Alt screenshot of reviewers being assigned](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/codeowners.png)

> Setting up code owners is easy, check [this guide](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) to get started with code owners.

Reviewer 2: The use of `code owners` here should also be consistent with the lines above.

### Let a panda spam your co-workers into submission for reviews.

> One thing that helped our team a lot was the addition of [Pull panda](https://pullreminders.com/). It is a Slack integration that automatically sends a message to your co-workers when someone requests their review. 

Reviewer 2: The mixing of past and present tense is slightly jarring to me. I would suggest:
```suggestion
One thing that has been very helpful to our team is the addition of [Pull panda](https://pullreminders.com/). It is a Slack integration that automatically sends a message to your co-workers when someone requests their review. 
```

When you request a review, the reviewer gets a message like:

![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-assigned.png)

If someone approves your pull request, you'll be notified as well:

![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-approved.png)

Even if someone comments on your pull request, panda will show you the comments right away in Slack:

![Alt screenshot of pull panda request message](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/panda-comments.png)

### Automatically merge the pull request

> Our pull request got labeled automatically, the correct reviewers got assigned automatically and they got a Slack message reminding them that there is work to be done. Now there is one last thing ~~we~~ the robots should do, merge the pull request into the master branch! If your team uses any form of automated checks on pull requests, you have probably already wasted some time blindly staring at your screen until the merge button turns green.

Reviewer 2: Same here. I would suggest:
```suggestion
Our pull requests get labeled automatically, the correct reviewers get assigned automatically and they get a Slack message reminding them that there is work to be done. Now there is one last step left over for ~~us~~ the robots to do, merge the pull request into the master branch! If your team uses any form of automated checks on pull requests, you have probably already wasted some time blindly staring at your screen until the merge button turns green.
```

Author: The past tense is deliberate here. I am trying to build a narrative where the user creates a "virtual pull request". But maybe I can reword this a bit.

> We can even automate the merging of our pull request with another GitHub action called [automerge](https://github.com/pascalgn/automerge-action). This action automatically merges your pull request based on your team's preferences.

Reviewer 2: Even is superfluous here because you already introduced this concept in the previous sentence.

![Alt screenshot of GitHub merging the pull request](https://raw.githubusercontent.com/MyOnlineStore/blogs/how-to-go-home-early/public/automerge.png)

> We configured it so that it merges a pull request once:

Reviewer 2: This sentence is a bit dutch to my ears. I would replace it with something like:
```suggestion
We set up automerge to merge a pull request once:
```
